Home work2

Shtuka1993 opened the pull request:
Home Work Two

yshtokaylo committed:
Home work two.

yshtokaylo committed:
Fixed SnakeCaseToCamelTest for loop.

yshtokaylo committed:
Fixed mirrorMultibyteString. Created own function mb_strrev for reverting multibyte strings.

REM: After fetch there was update in composer. It include related files. It is necessary for UnitTest. Do not forget to update composer to include them.

modanets committed:
updated instructions

yderevianko-cartera commented on src/02-strings.php:
```diff
@@ -9,6 +9,18 @@
 {
+    $result = $input;
+    $length = strlen($input);
+    for ($i=0; $i<$length; $i++) {
```
PSR spaces

Shtuka1993 replied:
Оновив.

yderevianko-cartera commented on src/02-strings.php:
```diff
@@ -9,6 +9,18 @@
+            $oldLeter = $input[$i+1];
+            $newLeter = strtoupper($oldLeter);
+            $result = substr_replace($result, $newLeter, $i+1, 1);
+        }
```
Робичий спосіб, але ще можеш глянути до вбудованих php функцій для роботи з масивами і стрічками і два найпопулярніші з них: explode, implode.

Приклад можливого варіанта з використанням вбудованих функцій (це не обовязково так робити):


```
function snakeCaseToCamelCase(string $input)
{
    $inputArray = explode('_', $input);
    $inputArray = array_map(
        function ($elem) {
            return ucfirst($elem);
        },
        $inputArray
    );

    return lcfirst(implode('', $inputArray));
}

```

Shtuka1993 replied:
Дякую. Оновлю код.

Shtuka1993 replied:
Дякую. Спробую реалізувати.

Shtuka1993 replied:
Я додав цей варіант собі в код, та все ж зберіг старий код.

yderevianko-cartera commented on src/02-strings.php:
```diff
@@ -21,6 +33,24 @@ function snakeCaseToCamelCase(string $input)
+function mb_strrev(string $word)
+{
+    $result = '';
+    $length = mb_strlen($word) - 1;
```
Хороший варіант )) 
Тільки власні (не вбудовані функції) потрібно називати camelCase і перед кожним return потрібно ставити пустий рядок.

Shtuka1993 replied:
Оновив.

yderevianko-cartera commented on src/02-strings.php:
```diff
@@ -39,4 +69,11 @@ function mirrorMultibyteString(string $input)
  */
 function getBrandName(string $noun)
 {
+    $result = '';
```
Не потрібна змінна так як вона завжди буде ініціалізована в if\else

Також потрібно ставити пробіл до і після '.'

Shtuka1993 replied:
Оновив.

modanets committed:
arrays

yshtokaylo committed:
Updates on comments.
Fixed PSR styles, renamed custom function, added blank lines before return statements.

yshtokaylo committed:
Merge branch 'master' into HomeWork2

yshtokaylo committed:
Added recommended code for snakeCaseToCamelCase function. Saved old version fo code in snakeCaseToCamelCaseOld function.
